### ISSUE TITLE
macOS code signing failing because "larger updated load commands do not fit (the program must be relinked using a larger -headerpad value)"

### ISSUE THREAD

**Issue by 4ian** [ceifa/steamworks.js#103]
Hi,

A pull request was submitted to integrate steamwork.js into GDevelop (open-source game engine) by @arthuro555. See https://github.com/4ian/GDevelop/pull/5440, based on https://github.com/4ian/GDevelop/pull/5401.

Unfortunately, the code-signing of GDevelop desktop app (based on Electron, built with electron-builder) is failing when steamworks.js is in the node_modules:
https://app.circleci.com/pipelines/github/4ian/GDevelop/10214/workflows/ac492663-bbb3-4845-8cc8-8f6a0832599f/jobs/11306

```
fatal error: /Applications/Xcode-12.5.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/codesign_allocate: can't allocate code signature data for: /Users/distiller/project/newIDE/electron-app/dist/mac/GDevelop 5.app/Contents/Resources/app.asar.unpacked/node_modules/steamworks.js/dist/osx/steamworksjs.darwin-x64.node (for architecture x86_64) because larger updated load commands do not fit (the program must be relinked using a larger -headerpad value)
/Users/distiller/project/newIDE/electron-app/dist/mac/GDevelop 5.app/Contents/Resources/app.asar.unpacked/node_modules/steamworks.js/dist/osx/steamworksjs.darwin-x64.node: the codesign_allocate helper tool cannot be found or used
  failedTask=build stackTrace=Error: Command failed: codesign --sign FB6D820A03351D2A14358DEB71ACF68A9ACC2CCE --force --keychain /var/folders/bl/wbxjgtzx7j5_mjsmfr3ynlc00000gp/T/f0dd2d22bf3ebf95800bc2cac4ce74a5430f4d93ec74f5e0171b03fbbf715269.keychain --timestamp --options runtime --entitlements /Users/distiller/project/newIDE/electron-app/build/entitlements.mac.inherit.plist /Users/distiller/project/newIDE/electron-app/dist/mac/GDevelop 5.app/Contents/Resources/app.asar.unpacked/node_modules/steamworks.js/dist/osx/steamworksjs.darwin-x64.node
fatal error: /Applications/Xcode-12.5.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/codesign_allocate: can't allocate code signature data for: /Users/distiller/project/newIDE/electron-app/dist/mac/GDevelop 5.app/Contents/Resources/app.asar.unpacked/node_modules/steamworks.js/dist/osx/steamworksjs.darwin-x64.node (for architecture x86_64) because larger updated load commands do not fit (the program must be relinked using a larger -headerpad value)
/Users/distiller/project/newIDE/electron-app/dist/mac/GDevelop 5.app/Contents/Resources/app.asar.unpacked/node_modules/steamworks.js/dist/osx/steamworksjs.darwin-x64.node: the codesign_allocate helper tool cannot be found or used
              
```

I read "the program must be relinked using a larger -headerpad value". Is this something that can be done on the steamworkjs.darwin-x64.node binary?

**Comment by ceifa** [comment]
It can be a regression from #92
Can you try build with the version `0.1.0`, I don't have a Mac to try :(

**Comment by Silver-Streak** [comment]
Hey @ceifa 

I'm one of the folks who commissioned integrating this into GDevelop. It appears they've found some workarounds/fixes at https://github.com/arthuro555/steamworks.js/commits/main (Starting July 19th). 

Does this fix sound valid for the main project as a PR? I'd hate for multiple forks to exist just to get it into GDevelop, and it seems like it might benefit everyone using the main steamworks.js.

**Comment by arthuro555** [comment]
The problem seems to be linked to using zig as the linker. Ideally, we should try to report this higher up the chain (zig, rust napi?). For now, doing Mac builds on a Mac machine and removing the zig flag allows builds to be correctly linked.

**Comment by ceifa** [comment]
Glad to hear that you guys found a fix that can be used on github actions! Of course this is valid PR @Silver-Streak
Thanks for it!

**Comment by ceifa** [comment]
released on 0.3.1
